components/script/build.rs

```diff
@@ -37,7 +37,12 @@ fn main() {
         // for reasons that I don't understand.  If we just give
         // link.exe, it tries to use script-*/out/link.exe, which of
         // course does not exist.
-        build.define("CMAKE_LINKER", "C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\bin\\amd64\\link.exe");
+        let platform = env::var("PLATFORM").expect("Environment variable 'PLATFORM' is not defined");
+        let link_path = PathBuf::from(env::var("VCINSTALLDIR").unwrap())
+                                .join("bin")
+                                .join(if platform.to_uppercase() == "X64" { "amd64" } else { "" })
```

Comment:
nit: You could use `AsciiExt` and do `if platform.eq_ignore_ascii_case("x64")`, but not that it matters a lot.